### ISSUE TITLE
alternative to port name prefix for protocol labelling (was alt updating portname on service, openshift routes)

### ISSUE THREAD

**Issue by debianmaster** [istio/istio#415]
is there any other way i can patch service port name to make it ready for istio  instead of naming it as `http-` ?
some  times service port names are already used by something else and changing the name will make dependent object go crazy.   

Ex:-   openshift uses service port-name to map inside routes, changing the portname will affect dependent routes  cc / @smarterclayton

**Comment by smarterclayton** [comment]
I'm not sure what the question is - openshift routes can point directly to a backend endpoint if necessary, so you can change from name to the port number on the container before you change the service port name, then change back.

**Comment by debianmaster** [comment]
@smarterclayton 
Background:   istio needs port name of a service to updated to format like `http-`  , currently openshift creates `something like 8080-tcp`    by default

yes we can update  openshift route object to point to new port name of the service that totally works.  just see if there is any metadata section on service ports section that can be used by istio instead of relying on port name.  

TL;DR not an openshift issue but looking at options Istio folks may have more info.

**Comment by smarterclayton** [comment]
Depending on a specific port name is somewhat sketchy :)  This is kind of a
gap in Kube, that there is no port metadata, so I do understand picking a
heuristic.  I'd suggest that an annotation be available to pick an
appropriate port if the port name doesn't match.

**Comment by smarterclayton** [comment]
8080-tcp is the default from Kubernetes, not OpenShift.

**Comment by andraxylia** [comment]
The name prefix is used to infer the protocol:
https://github.com/istio/pilot/blob/master/platform/kube/conversion.go#L103

Feel free to propose a different model and submit a PR.

**Comment by ldemailly** [comment]
we should probably support explicit labels/annotations

putting this on for 0.3

**Comment by costinm** [comment]
Closing for now, it doesn't look like we have anyone contributing on this.

**Comment by ldemailly** [comment]
This is still needed

**Comment by debianmaster** [comment]
This issue still persists in openshift.

**Comment by debianmaster** [comment]
This issue still persists in openshift.
